lib/diaspora_federation/entity.rb

```diff
@@ -264,21 +315,52 @@ def simple_node(doc, name, value)
       end
     end
 
-    # @param [Nokogiri::XML::Element] root_node xml nodes
+    def json_data
+      enriched_properties.map {|key, value|
+        type = self.class.class_props[key]
+
+        if type.instance_of?(Class) && value.respond_to?(:to_json_hash)
+          entity_data = value.to_json_hash[:entity_data]
+          [key, entity_data] unless entity_data.nil?
+        else
+          [key, value]
+        end
```

Comment:
In the version that I'm preparing now I followed the idea that `to_json` returns a hash, which should be transformed to JSON by user (like it works for `to_xml` where we get `Nokogiri::XML::Element` and not a String with XML). 

So in this very case we don't need ActiveSupport, unless we wan't to transform actual `to_json` output (a hash) to a string with JSON representation.